internal/generator/generator_util.go

```diff
@@ -201,3 +228,52 @@ func genFromSrcCode(codePath string, mapVariableName string, matchFnName string,
 
 	return nil
 }
+
+func strList2Map(strs []string) map[string]struct{} {
+	if strs == nil {
+		return nil
+	}
+
+	rv := map[string]struct{}{}
+	for _, str := range strs {
+		rv[str] = struct{}{}
+	}
+
+	return rv
+}
```

Comment:
You created a special `[]string` type to collect filters, could that have been a `map[string]struct{}` to begin with?

```go
type AllowedDirectives struct {
	directiveNames map[string]struct{}
}

func (a *AllowedDirectives) Set(s string) error {
	if a.directiveNames == nil {
		a.directiveNames := make(map[string]struct{})
	}
	a.directiveNames[s] = struct{}{}
	return nil
}
``